### PR TITLE
Add I2S/SAI pad info

### PR DESCRIPTION
**Pull request by lkolbly**
This adds traits and impls for the SAI peripherals. (also, it updates the Cargo.toml to point at this repo)

This is very 106x-specific, should I wrap it in a `cfg(feature = "imxrt106x")`? That's the only feature in the crate (on the `v0.1` branch), so I'm not sure. The 1010 I believe does have some (2?) SAI peripherals, and the pads are almost certainly not the same.

SAI3 overlaps on several pins with SAI1, so adding it complains about re-implementing a few traits. There may be a clever way around this (in particular, the traits could be split into separate e.g. `RxSyncPin` and `RxBclkPin` traits), I can do that if we want but I don't feel a pressing need to do so right now. Also the daisy names are different for some reason.

Testing-wise, I've only tested this with SAI1 in one-bit TX mode, with a specific set of pins. I tried to copy the table as closely as possible.

### CODE REVIEW

**Comment by bfrog** [comment]
Really happy to see this happening!

File: Cargo.toml
Comment (bfrog) [inline Cargo.toml:10]: 👍

File: daisy.py
Comment (bfrog) [inline daisy.py:8]: :+1:

**Review by bfrog — COMMENTED** [review]
Looks pretty good generally, I need to look closer at the SAI pads and alts from the reference doc

**Comment by lkolbly** [comment]
> I need to look closer at the SAI pads and alts from the reference doc

Please do and take your time, the 3s and the 8s blend together on my screen. :D

**Review by mciantyre — COMMENTED** [review]
Great start! A few questions and requests.

Can we target the main branch for this PR? We don't have to add 1010 SAI pins right now, but we should design knowing that those pins will eventually be implemented. I'll take care of backporting to the 0.1 branch.

> SAI3 overlaps on several pins with SAI1, so adding it complains about re-implementing a few traits.

The ADC pins had the same issue. Check out the `adc` module to see how we solved it: the `Pin` trait is generic over an ADC module. Instead of using custom types for SAI[1, 2, 3] (analogous to `ADC1` and `ADC2`), let's try using `typenum` types to express the SAI instance. If my thinking is correct, it might work nicely with changes we're considering in the RAL.

> There may be a clever way around this (in particular, the traits could be split into separate e.g. RxSyncPin and RxBclkPin traits)

At the moment, I lean towards a single `Pin` trait to maintain consistency with the rest of the crate. But if we find that multiple traits is cleaner or easier for the user, I'm happy to support it.

Auto-generating the SAI implementations in `build.rs` diverges with the rest of the pin implementations. I love how the tables are easy to read and review, but I'm not sure it warrants the complexity of build-time generated code. Could we compromise with declarative macros to implement the pins? If auto-generated code is the way to go, let's move the types and routines into the `imxrt-iomuxc-build` crate, which will make it easier to test (now or later).

I took your branch and started playing around with these ideas. Check out the `sai-i2s` branch if you want to see what it looks like.

File: src/sai.rs
Comment (lkolbly) [reply src/sai.rs:55]: > I'm guessing that the daisy register applies to both

I don't think so, since I think the daisy registers only apply to inputs so they wouldn't apply to TX.

`RxDataPin` does have daisy registers though, good catch. I think it has one for every pin.

File: src/sai.rs
Comment (lkolbly) [reply src/sai.rs:48]: Hmm, I'm not sure you can. Ideally, I think the HAL implementation would look something like this:
```
    pub fn build_1bit_tx<MCLK, BCLK, SYNC, DATA>(
        self,
        clock_config: ClockConfig,
        mut mclk: MCLK,
        mut bclk: BCLK,
        mut sync: SYNC,
        mut data: DATA,
    ) -> SAI<M>
    where
        MCLK: sai::Pin<Module = M, Signal = sai::MCLK2>,
        BCLK: sai::Pin<Module = M, Signal = sai::TX_BCLK>,
        SYNC: sai::Pin<Module = M, Signal = sai::TX_SYNC>,
        DATA: sai::TxDataPin<Module = M, Index = U0>,
```
And in particular, the data pin arguments should only accept specific indexes, and I can't get the type bound to work with associated constants.

**Comment by lkolbly** [comment]
> I took your branch and started playing around with these ideas. Check out the sai-i2s branch if you want to see what it looks like.

Awesome, that branch looks pretty reasonable to me! I agree that the macro is cleaner than a build script.

I think we will need associated types for the indexes though, rather than consts.

File: src/sai.rs
Comment (lkolbly) [reply src/sai.rs:71]: Hm, I have not tested SAI input (or even multi-line output), so I'm not really in a position to easily test it. It's not clear to me whether that behaviour would transfer over to the 106x series, since ostensibly the 1010 has no daisy chain registers so the configuration scheme could be different.

File: src/sai.rs
Comment (mciantyre) [reply src/sai.rs:48]: Looks good to me! Yea, I hit a similar limitation when trying to move this entire crate to const generics. Let's stick with the typenums.

File: src/sai.rs
Comment (mciantyre) [reply src/sai.rs:71]: Let's keep this as-is. Once merged, we'll track our uncertainty / future testing with an issue.

**Comment by mciantyre** [comment]
Agreed on keeping typenums for data line indices! Pushed a corresponding commit to the `sai-i2s` branch in this remote.

File: src/sai.rs
Comment (bfrog) [reply src/sai.rs:48]: Type bounds are a very good reason to keep the typenums, I agree!

**Review by mciantyre — APPROVED** [review]
LGTM! Thanks for working through this. I'll leave this open for @bfrog to take a look. Unless we need more time to review, I'll merge after a week, and release 0.1.3 with these changes.

**Review by bfrog — APPROVED** [review]
This looks pretty darn solid to me at this point, again I didn't look too deeply at the pad and iomux values, but more broadly at the macro, api, and such

**Pull request merged**
